### PR TITLE
MINIFICPP-1232 - PublishKafka processor doesn't validate some properties

### PR DESCRIPTION
**Pull request by adam-markovics**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [x] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE file?
- [x] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/librdkafka/PublishKafka.cpp [hunk 1]
```diff
@@ -82,10 +82,33 @@ core::Property PublishKafka::TargetBatchPayloadSize(
     core::PropertyBuilder::createProperty("Target Batch Payload Size")->withDescription("The target total payload size for a batch. 0 B means unlimited (Batch Size is still applied).")
         ->isRequired(false)->withDefaultValue<core::DataSizeValue>("512 KB")->build());
 core::Property PublishKafka::AttributeNameRegex("Attributes to Send as Headers", "Any attribute whose name matches the regex will be added to the Kafka messages as a Header", "");
-core::Property PublishKafka::QueueBufferMaxTime("Queue Buffering Max Time", "Delay to wait for messages in the producer queue to accumulate before constructing message batches", "");
-core::Property PublishKafka::QueueBufferMaxSize("Queue Max Buffer Size", "Maximum total message size sum allowed on the producer queue", "");
-core::Property PublishKafka::QueueBufferMaxMessage("Queue Max Message", "Maximum number of messages allowed on the producer queue", "");
-core::Property PublishKafka::CompressCodec("Compress Codec", "compression codec to use for compressing message sets", COMPRESSION_CODEC_NONE);
+
+const core::Property PublishKafka::QueueBufferMaxTime(
```
Comment (adamdebreceni) [inline extensions/librdkafka/PublishKafka.cpp:86]: to nitpick: with method chaining I don't know what the style guideline is, but I prefer the operator on the next line (and it seems that other properties do it like this, in this file)

Comment (arpadboda) [reply extensions/librdkafka/PublishKafka.cpp:86]: +1.

That's the way linter accepts it as well, so please change as @adamdebreceni suggest!

File: extensions/librdkafka/PublishKafka.h [hunk 2]
```diff
@@ -89,10 +89,10 @@ class PublishKafka : public core::Processor {
...
   static core::Property TargetBatchPayloadSize;
   static core::Property AttributeNameRegex;
-  static core::Property QueueBufferMaxTime;
-  static core::Property QueueBufferMaxSize;
-  static core::Property QueueBufferMaxMessage;
-  static core::Property CompressCodec;
+  static const core::Property QueueBufferMaxTime;
+  static const core::Property QueueBufferMaxSize;
+  static const core::Property QueueBufferMaxMessage;
+  static const core::Property CompressCodec;
```
Comment (arpadboda) [inline extensions/librdkafka/PublishKafka.h:95]: Either all or none pls!

File: extensions/librdkafka/PublishKafka.cpp (hunk 1 above)
Comment (hunyadi-dev) [reply extensions/librdkafka/PublishKafka.cpp:86]: I like it this way, so 2/2 :)

Comment (arpadboda) [reply extensions/librdkafka/PublishKafka.cpp:86]: Not sure if this still applies after your linter changes, but previously the linter only accepted the "->" operators to be at the begging of the new line. 
If you take a look at standard processors, all use the other format.

Comment (adam-markovics) [reply extensions/librdkafka/PublishKafka.cpp:86]: Done in new commit.

File: extensions/librdkafka/PublishKafka.h (hunk 2 above)
Comment (adam-markovics) [reply extensions/librdkafka/PublishKafka.h:95]: Done in new commit.

**Pull request closed without merging**